signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
That definitely needs to happen _somewhere_, and right now it doesn’t.  This is a bad fit, though, we should be verifying the user-specified digest even if no signatures exist or are required.  So far I was thinking that the individual `ImageSource`s which support digest references should do that in their `GetManifest`s; it _could_ probably be shared in `copy.Image`, OTOH it seems reasonable that this should be rejected for _any_ use of that `ImageSource`.